### ISSUE TITLE
Signals are not handler properly

### ISSUE THREAD

**Issue by int010h** [gchp/rustbox#53]
My term is completely broken after finishing application with SIGINT/SIGTERM (cursor and input characters are invisible, x axis positioning is broken too), I guess those signals should be handled to terminate resources properly.
I'm using Arch Linux and termite terminal emulator.

**Comment by steakknife** [comment]
Yup, tui programs almost always need to install signal handlers.

Run `stty` before and after to find the diff.

Quick fix is `reset`.
